### DISCUSSION TITLE
Any thoughts on getting pyinaturalist included with Pyodide?

### DISCUSSION THREAD

**Discussion by miistakis** [pyinat/pyinaturalist#523] [Ideas]
Just a thought, if its included with Pyodide we can use pyinaturalist with py-scripts as well and write code right in our HTML pages, which opens up a lot of options for displaying inat data back on our websites.

**Comment by JWCook** [comment]
Good question! I'd be interested to hear more about what you're working on.

My first thought is that iNaturalist's own javascript package ([inaturalistjs](https://github.com/inaturalist/inaturalistjs)) is going to be better suited for displaying iNat data on your own website.

Otherwise, pyinaturalist should theoretically be usable in Pyodide already. If you're referring to the [wheels built specifically for Pyodide](https://pyodide.org/en/stable/usage/packages-in-pyodide.html), I believe those are only necessary for packages that depend on C extensions, which need to be compiled for WASM.

For pure python packages (including this one), you can install with [micropip](https://pyodide.org/en/stable/usage/loading-packages.html). The only extra step needed is to explicitly install the `sqlite3` module (part of the standard library, but it looks like [Pyodide removed it in 0.22](https://pyodide.org/en/stable/project/changelog.html#version-0-22-0)):
```py
import micropip
await micropip.install('sqlite3', 'pyinaturalist')
```

**Reply by miistakis** [reply]
inaturalistjs does not have the same power, and as someone that only does enough python to get by, its nice to have your code base between your tools and your website to be same/similar 

I will see if I can install via micropip in the py-script code, thanks for the suggestion.
